debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java

```diff
@@ -218,10 +221,45 @@ protected void complete(SnapshotContext snapshotContext) {
      * @return a valid query string
      */
     @Override
-    protected Optional<String> getSnapshotSelect(SnapshotContext snapshotContext, TableId tableId) {
+    protected Optional<String> getSnapshotSelect(RelationalSnapshotContext snapshotContext, TableId tableId) {
+        String modifiedColumns = checkExcludedColumns(tableId);
+        if (modifiedColumns != null) {
+            return Optional.of(String.format("SELECT %s FROM [%s].[%s]", modifiedColumns, tableId.schema(), tableId.table()));
+        }
         return Optional.of(String.format("SELECT * FROM [%s].[%s]", tableId.schema(), tableId.table()));
     }
 
+    @Override
+    protected String enhanceOverriddenSelect(RelationalSnapshotContext snapshotContext, String overriddenSelect, TableId tableId) {
+        String modifiedColumns = checkExcludedColumns(tableId);
+        if (modifiedColumns != null) {
+            overriddenSelect = overriddenSelect.replaceAll("\\*", modifiedColumns);
+        }
+        return overriddenSelect;
+    }
+
+    private String checkExcludedColumns(TableId tableId) {
+        String modifiedColumns = null;
+        String excludedColumnStr = connectorConfig.getTableFilters().getExcludeColumns();
```

Comment:
I don't understand how this method is supposed to work. The excluded columns are not removed from the statement anywhere. Also note that the exclusion expression doesn't necessarily contain the table name (e.g. could be `.*mycolumn`). This entire method seems flawed to me.

That said, it doesn't seem to to harm either, so I'll for now simply move https://issues.redhat.com/browse/DBZ-2283 onwards to the next iteration.